crates/bevy_core_pipeline/src/prepass/mod.rs

```diff
@@ -0,0 +1,131 @@
+//! Run a prepass before the main pass to get the depth and/or normals texture.
+//! The depth prepass texture is then used by the main pass to reduce overdraw.
+//!
+//! To enable the prepass, you need to add a `PrepassSettings` component to a `Camera`.
+//! Both textures are available on the `PrepassTextures` component attached to each `Camera` with a `PrepassSettings`
+//!
+//! Currently only works for 3d
+
+pub mod node;
+
+use bevy_ecs::prelude::*;
+use bevy_render::{
+    render_phase::{CachedRenderPipelinePhaseItem, DrawFunctionId, EntityPhaseItem, PhaseItem},
+    render_resource::{CachedRenderPipelineId, Extent3d},
+    texture::CachedTexture,
+};
+use bevy_utils::FloatOrd;
+
+/// Add a `PrepassSettings` component to a `Camera` to perform a depth and/or normal prepass.
```

Comment:
The prepass could technically run in 2d too which is why I didn't specify a dimension here.

Reply:
What would the point of running a 2d prepass be? Also I'm pretty sure the prepass code checks that it's a Camera3d.

Reply:
Yes, the query in `extract_core_3d_camera_prepass_phase` has a `With<Camera3d>` but it only uses the `Camera`, so it could be used as a 2d prepass plugin too. You could easily remove the `With` and it would still work.

As to why you'd want a 2d prepass, not sure honestly, maybe for tilemaps with a bunch of sparse layers? Something like that could create overdraw I guess.

Anyway, the main reason I didn't put a dimension there was because the query only uses the `Camera` component.

Reply:
A prepass for 2D would also be to avoid overdraw with expensive shaders. Currently all 2D is rendered using alpha blending and no depth attachment, but people have been working on using PBR shading for 2D as well, and I've seen situations where use of a depth buffer would definitely be desirable to avoid overdraw.